lib/put.js

```diff
@@ -0,0 +1,175 @@
+const _ = require('lodash')
+const debug = require('debug')('signalk-server:put')
+
+const pathPrefix = '/signalk'
+const versionPrefix = '/v1'
+const apiPathPrefix = pathPrefix + versionPrefix + '/api/'
+
+const State = {
+  pending: 'PENDING',
+  completed: 'COMPLETED',
+  notSupported: 'NOT SUPPORTED'
+}
+
+const Result = {
+  success: 'SUCCESS',
+  failure: 'FAILURE'
+}
+
+const actionHandlers = {}
+const actions = {}
+var nextActionId = 1
+
+const pruneActionTimeout = 60 * 60 * 1000
+const pruneInterval = 15 * 60 * 1000
+
+module.exports = {
+  start: function (app) {
+    app.registerActionHandler = registerActionHandler
+    app.deRegisterActionHandler = deRegisterActionHandler
+
+    setInterval(pruneActions, pruneInterval)
+
+    app.get(apiPathPrefix + 'actions', function (req, res, next) {
+      res.json(actions)
+    })
+
+    app.get(apiPathPrefix + 'actions/:id', function (req, res, next) {
+      var action = actions[req.params.id]
+      if (!action) {
+        res.status(404).send()
+      } else {
+        res.json(action)
```

Comment:
Everybody can see all actions?

Reply:
Yes, currently anyone with read access can see all actions. Do you think we should do something else?

Reply:
I’m ok with that, especially with security by uuid obscurity.

Reply:
I’m ok with that, especially with security by uuid obscurity.